### ISSUE TITLE
PYTHONPATH is polluted

### ISSUE THREAD

**Issue by nicokist** [dnanexus/dx-toolkit#453]
currently dnanexus changes the PYTHONPATH when the `environment` file is sourced, causing buggy behaviour with non-dnanexus python programs. For instance, on Ubuntu misspellings on the command line now throw a python error instead of a helpful message. This happens because the Ubuntu tool sources the dnanexus enum package rather than the standard library enum. 

Instead of manipulating the standard PYTHONPATH would you consider something like the following?

```python
#!/usr/bin/env python2.7

# -*- coding: utf-8 -*-
import re
import sys
import os

sys.path.insert(0, os.environ['DNANEXUS_HOME']+'/share/dnanexus/lib/python2.7/site-packages')
sys.path.insert(0, os.environ['DNANEXUS_HOME']+'/lib/python')

from dxpy.scripts.dx import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

**Comment by orodeh** [comment]
You are correct, this is a known issue, which we are planning to ameliorate in the future. It is not easy to solve though. 

Our recommendation, in this case, is to run dxpy in a virtual environment, so it does not pollute your system. For example, you can run it from docker, pyenv, or conda.

**Comment by odoublewen** [comment]
This issue has vexed me in the past as well -- making it hard/impossible to use Py3 code that I've developed.  

 I now use this pattern as a workaround.  (I don't like that I have to do it, but it works fine.)  Assuming your app has a bash entrypoint, the `code.sh` script would look something like:

```bash
dx-download-all-inputs

# Set PYTHONPATH aside so we don't pollute our nice py3 environment
_PYTHONPATH=${PYTHONPATH}
unset PYTHONPATH
source /py3venv/.venv/bin/activate

#
# Call your Python3 code here
#

# Now deactivate the venv and reset PYTHONPATH so dx toolkit is happy
deactivate
export PYTHONPATH=${_PYTHONPATH}

dx-upload-all-outputs
```

(IMHO, [the time is nigh](https://python3statement.org/) for DNAnexus to support Python3 officially.  I can imagine it's a big push but it's an important one.)
